packages/wasm-dpp/src/data_contract/data_contract_facade.rs

```diff
@@ -0,0 +1,109 @@
+use crate::errors::protocol_error::from_protocol_error;
+use crate::{DataContractCreateTransitionWasm, DataContractUpdateTransitionWasm, DataContractWasm};
+use dpp::data_contract::DataContractFacade;
+use dpp::identifier::Identifier;
+use dpp::version::ProtocolVersionValidator;
+
+use crate::validation::ValidationResultWasm;
+use std::sync::Arc;
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen(js_name=DataContractFacade)]
+#[derive(Clone)]
+pub struct DataContractFacadeWasm(Arc<DataContractFacade>);
+
+impl DataContractFacadeWasm {
+    pub fn new(
+        protocol_version: u32,
+        protocol_version_validator: Arc<ProtocolVersionValidator>,
+    ) -> Self {
+        let inner = DataContractFacade::new(protocol_version, protocol_version_validator);
+
+        Self(Arc::new(inner))
+    }
+}
+
+#[wasm_bindgen(js_class=DataContractFacade)]
+impl DataContractFacadeWasm {
+    /// Create Data Contract
+    #[wasm_bindgen]
+    pub fn create(
+        &self,
+        owner_id: Vec<u8>,
+        documents: JsValue,
+    ) -> Result<DataContractWasm, JsValue> {
+        let id = Identifier::from_bytes(&owner_id).map_err(from_protocol_error)?;
+
+        self.0
+            .create(id, serde_wasm_bindgen::from_value(documents)?)
+            .map(Into::into)
+            .map_err(from_protocol_error)
+    }
+
+    /// Create Data Contract from plain object
+    #[wasm_bindgen(js_name=createFromObject)]
+    pub async fn create_from_object(
+        &self,
+        js_raw_data_contract: JsValue,
+        skip_validation: bool,
+    ) -> Result<DataContractWasm, JsValue> {
+        let raw_data_contract = serde_wasm_bindgen::from_value(js_raw_data_contract)?;
+        self.0
+            .create_from_object(raw_data_contract, skip_validation)
+            .await
+            .map(Into::into)
+            .map_err(from_protocol_error)
+    }
+
+    /// Create Data Contract from buffer
+    #[wasm_bindgen(js_name=createFromBuffer)]
+    pub async fn create_from_buffer(
+        &self,
+        buffer: Vec<u8>,
+        skip_validation: bool,
```

Comment:
to keep the interface compatible  with JS and make  `skipValidation` optional you could use something like this:
https://github.com/dashpay/platform/blob/2ec9eb16d9495914ac3aee3a1f4752fa50710428/packages/wasm-dpp/src/document/state_transition/document_batch_transition/mod.rs#L157